presto-main/src/main/java/io/prestosql/sql/planner/DistributedExecutionPlanner.java

```diff
@@ -180,16 +186,18 @@ private Visitor(Session session, StageExecutionDescriptor stageExecutionDescript
                     .map(DynamicFilters.ExtractResult::getDynamicConjuncts)
                     .orElse(ImmutableList.of());
 
-            // TODO: Execution must be plugged in here
-            if (!dynamicFilters.isEmpty()) {
+            Supplier<TupleDomain<ColumnHandle>> dynamicFilterSupplier = TupleDomain::all;
+            if (!dynamicFilters.isEmpty() && !stageExecutionDescriptor.isScanGroupedExecution(node.getId())) {
```

Comment:
I don't think `!stageExecutionDescriptor.isScanGroupedExecution(node.getId())) {` condition is required here. It's more for join so that it executed fixed amount of times, but we already have check in `LocalExecutionPlanner.Visitor#createDynamicFilter`

Reply:
Yes, I've removed it now